internal/controller/datadogagent/override/podtemplatespec_test.go

```diff
@@ -204,64 +204,6 @@ func TestPodTemplateSpec(t *testing.T) {
 				)
 			},
 		},
-		{
-			name: "08: given name:tag, override image with full URI name, ignore tag and JMX, full name takes precedence",
-			existingManager: func() *fake.PodTemplateManagers {
-				return fakePodTemplateManagersWithImageOverride(containerImageOptions{name: "agent:7.38.0"}, t)
-			},
-			override: v2alpha1.DatadogAgentComponentOverride{
-				Image: &v2alpha1.AgentImageConfig{
-					Name:       "someregistry.com/datadog/agent:9.99.9",
-					Tag:        "latest",
-					JMXEnabled: true,
-				},
-			},
-			validateManager: func(t *testing.T, manager *fake.PodTemplateManagers) {
-				assertImageConfigValues(
-					manager,
-					containerImageOptions{name: "someregistry.com/datadog/agent:9.99.9"},
-					t,
-				)
-			},
-		},
-		{
-			name: "09: given name:tag, override image name, tag, JMX, sets default registry",
-			existingManager: func() *fake.PodTemplateManagers {
-				return fakePodTemplateManagersWithImageOverride(containerImageOptions{name: "agent:7.38.0"}, t)
-			},
-			override: v2alpha1.DatadogAgentComponentOverride{
-				Image: &v2alpha1.AgentImageConfig{
-					Name:       "agent",
-					Tag:        "latest",
-					JMXEnabled: true,
-				},
-			},
-			validateManager: func(t *testing.T, manager *fake.PodTemplateManagers) {
-				assertImageConfigValues(
-					manager,
-					containerImageOptions{name: "gcr.io/datadoghq/agent:latest-jmx"},
-					t,
-				)
-			},
-		},
-		{
-			name: "10: given name:tag, override image with name:tag",
-			existingManager: func() *fake.PodTemplateManagers {
-				return fakePodTemplateManagersWithImageOverride(containerImageOptions{name: "agent:7.38.0"}, t)
-			},
-			override: v2alpha1.DatadogAgentComponentOverride{
-				Image: &v2alpha1.AgentImageConfig{
-					Name: "agent:latest",
-				},
-			},
-			validateManager: func(t *testing.T, manager *fake.PodTemplateManagers) {
-				assertImageConfigValues(
-					manager,
-					containerImageOptions{name: "agent:latest"},
-					t,
-				)
-			},
-		},
```

Comment:
Removed a few test cases because these situations (name:tag set for the image) cannot occur with the current flow of code.